src/Validators/validatePositiveNumber.ts

```diff
@@ -0,0 +1,9 @@
+import { NegativeNumberError } from "../Errors";
+
+const validatePositiveNumber = (someNumber: number) => {
```

Comment:
los metodos que validan numbers estaria bueno que tomaran number | string y que hagan un Number(value) al principio

digo porque en el front el value de los textfield viene en string, y eso lo haria mucho mas comodo

tal vez validaciones que extiendan la anterior, tipo

validateNumber -> chequea que Number(value) no sea NaN
validateInteger -> validateNumber && Number.isInteger(Number(value))
validatePositive -> validateNumber && Number(value) > 0

Reply:
y tal vez un

validatePositiveInteger -> validateInteger && validatePositive

que creo que es lo que buscamos

Reply:
Pero queres que separa eso en en tres validators distintos?

Reply:
Y por que que reciba number | string? no podria hacerse desde afuera un Number(value) ?

Reply:
yo tendria los 4 validators para componer el validatePositiveInteger, así la lógica puede ser modular

y sobre lo del Number(value), si... pero en el front habría que hacer eso todo el tiempo por cada campo numerico... o sea prefiero directamente interpretar que "12" es un valid positive integer

Reply:
Si, entiendo

Reply:
![image](https://user-images.githubusercontent.com/20487262/78632455-6e953380-7875-11ea-8b56-859ce042c324.png)

igual tenemos esto... podemos mandarle number: true y hacerlo ahi

Reply:
me resumo: me pareceria copado tener un

validateNumber -> chequea que Number(value) no sea NaN
validateInteger -> validateNumber && Number.isInteger(Number(value))
validatePositive -> validateNumber && Number(value) > 0
validatePositiveInteger -> validateInteger && validatePositive

lo demas flashee

Reply:
validateNumber deberia llamarse validateNotNan quizas jaja. un number es un integer en typescript

Reply:
Bueno aca lo hace jaja
https://stackoverflow.com/questions/42654595/how-to-check-if-value-is-nan